Assets/Scripts/Augment/GunFactory.cs

```diff
@@ -18,9 +18,56 @@ public static GameObject InstantiateGun(Item bodyPrefab, Item barrelPrefab, Item
         // Initialize everything
         gun.GetComponent<GunFactory>().InitializeGun(owner);
 
+        var cullingLayer = LayerMask.NameToLayer("Gun " + owner.inputManager.playerInput.playerIndex);
+        SetGunLayer(gun.GetComponent<GunFactory>(), cullingLayer);
+
+        GunFactory displayGun = owner.GunOrigin.GetComponent<GunFactory>();
+        displayGun.Body = bodyPrefab;
+        displayGun.Barrel = barrelPrefab;
+        displayGun.Extension = extensionPrefab;
+        displayGun.InitializeGun();
+
+        var cullingLayerDisplay = LayerMask.NameToLayer("Player " + owner.inputManager.playerInput.playerIndex);
+        SetGunLayer(displayGun, cullingLayerDisplay);
+
+        var firstPersonGunController = gun.GetComponent<GunFactory>().GunController;
+        var gunAnimations = displayGun.GetComponentsInChildren<AugmentAnimator>(includeInactive: true);
+        foreach (var animation in gunAnimations)
+        {
+            animation.OnInitialize(firstPersonGunController.stats);
+            firstPersonGunController.onFire += animation.OnFire;
+            firstPersonGunController.onReload += animation.OnReload;
+        }
+
+        firstPersonGunController.RightHandTarget = displayGun.GunController.RightHandTarget;
+
+        if (displayGun.GunController.HasRecoil)
+            firstPersonGunController.onFire += displayGun.GunController.PlayRecoil;
+
+        if (displayGun.GunController.projectile.GetType() == typeof(BulletController))
+        {
+            ((BulletController) gun.GetComponent<GunFactory>().GunController.projectile).Trail.layer = 0;
+            firstPersonGunController.onFire += ((BulletController)displayGun.GunController.projectile).PlayMuzzleFlash;
+        }
+
+        if (displayGun.GunController.projectile.GetType() == typeof(MeshProjectileController))
+            ((MeshProjectileController) gun.GetComponent<GunFactory>().GunController.projectile).Vfx.gameObject.layer = 0;
```

Comment:
We don't have a way to just disable all the VFX calculations for the third-person guns?

Reply:
This function is actually the opposite: letting all players see the first-person mesh projectiles, disabling the calculation of third person gun projectiles happens another place. This way we only have to calculate the first person projectiles as it's oone of the most expensive calls we have

Reply:
but... does setting layer to 0 actually disable them?

Reply:
No, it ensures that no players cull that mesh projectile

Reply:
Oh, I didn't notice that it is set on the _actual_ gun and not the third-person display gun :sweat_smile: